src/tests/ftest/server/metadata.py

```diff
@@ -358,72 +354,71 @@ def test_metadata_server_restart(self):
-        self.out_queue = queue.Queue()
 
         processes = self.params.get("slots", "/run/ior/clientslots/*")
 
         list_of_uuid_lists = [
             [str(uuid.uuid4()) for _ in range(files_per_thread)]
             for _ in range(total_ior_threads)]
 
+        # Setup the thread manager
+        thread_manager = ThreadManager(run_ior_loop, self.timeout - 30)
+
         # Launch threads to run IOR to write data, restart the agents and
         # servers, and then run IOR to read the data
         for operation in ("write", "read"):
             # Create the IOR threads
-            threads = []
             for index in range(total_ior_threads):
-                # Define the arguments for the ior_runner_thread method
+                # Define the arguments for the run_ior_loop method
                 ior_cmd = IorCommand()
                 ior_cmd.get_params(self)
                 ior_cmd.set_daos_params(self.server_group, self.pool)
                 ior_cmd.flags.value = self.params.get(
                     "F", "/run/ior/ior{}flags/".format(operation))
 
                 # Define the job manager for the IOR command
-                manager = Orterun(ior_cmd)
-                env = ior_cmd.get_default_env(str(manager))
-                manager.assign_hosts(self.hostlist_clients, self.workdir, None)
-                manager.assign_processes(processes)
-                manager.assign_environment(env)
+                self.ior_managers.append(Orterun(ior_cmd))
+                self.ior_managers[-1].tmpdir_base.update(self.test_dir, "orterun.tmpdir_base")
+                env = ior_cmd.get_default_env(str(self.ior_managers[-1]))
+                self.ior_managers[-1].assign_hosts(self.hostlist_clients, self.workdir, None)
+                self.ior_managers[-1].assign_processes(processes)
+                self.ior_managers[-1].assign_environment(env)
+                self.ior_managers[-1].verbose = False
 
                 # Add a thread for these IOR arguments
-                threads.append(
-                    threading.Thread(
-                        target=ior_runner_thread,
-                        kwargs={
-                            "manager": manager,
-                            "uuids": list_of_uuid_lists[index],
-                            "results": self.out_queue}))
-
+                thread_manager.add(manager=self.ior_managers[-1], uuids=list_of_uuid_lists[index])
                 self.log.info(
                     "Created %s thread %s with container uuids %s", operation,
                     index, list_of_uuid_lists[index])
 
             # Launch the IOR threads
-            if self.thread_control(threads, operation) == "FAIL":
-                self.d_log.error("IOR {} Thread FAIL".format(operation))
-                self.fail("IOR {} Thread FAIL".format(operation))
+            self.log.info("Launching %d IOR %s threads", thread_manager.qty, operation)
+            failed_thread_count = thread_manager.check_run()
```

Comment:
check_run() returns boolean not a number, or am I missing something? Maybe just leave it as boolean for now and could consider a later enhancement (if it's deemed worth it) to return and report a count of threads that failed?

Reply:
I had made the change to `ThreadManager.check(0/.check_run()` and did not save the file so it was missing from the commit.  This has been fixed.